include/cuda/std/detail/libcxx/include/cmath

```diff
@@ -318,12 +318,76 @@ long double    truncl(long double x);
 #define NAN      __builtin_nan()
 #endif
 
-_LIBCUDACXX_BEGIN_NAMESPACE_STD
+// Forward to these functions to circumvent macro hackery in some compilers
+namespace {
+    template <typename _Ty>
+    _LIBCUDACXX_INLINE_VISIBILITY _Ty __libcudacxx_abs(_Ty __v) {
+#if defined(__cuda_std__) && defined(_LIBCUDACXX_COMPILER_GCC)
+        return ::std::abs(__v);
```

Comment:
Going to try an offer an explanation for the changes in this file:

GCC-5 and 6, do not offer `abs` in the global namespace. Everything else seems to or provides a macro that does.

Specifying `::abs` could break on systems that define a macro.

Reply:
In case it changes anything, the minimum gcc version we support moving forward is 7.

Reply:
This is resolved luckily by just getting rid of GCC5. :)